server/src/websocket_uploader/engine_uploader.c

```diff
@@ -0,0 +1,227 @@
+/*
+ *
+ * Copyright (C) 2022 University of Bamberg, Software Technologies Research Group
+ * <https://www.uni-bamberg.de/>, <http://www.swt-bamberg.de/>
+ * 
+ * This file is part of the SWTbahn command line interface (swtbahn-cli), which is
+ * a client-server application to interactively control a BiDiB model railway.
+ *
+ * swtbahn-cli is licensed under the GNU GENERAL PUBLIC LICENSE (Version 3), see
+ * the LICENSE file at the project's top-level directory for details or consult
+ * <http://www.gnu.org/licenses/>.
+ *
+ * swtbahn-cli is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or any later version.
+ *
+ * swtbahn-cli is a RESEARCH PROTOTYPE and distributed WITHOUT ANY WARRANTY, without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE. See the GNU General Public License for more details.
+ *
+ * The following people contributed to the conception and realization of the
+ * present swtbahn-cli (in alphabetic order by surname):
+ *
+ * - Bernhard Luedtke <https://github.com/bluedtke>
+ *
+ */
+
+#include "engine_uploader.h"
+#include "../server.h"
+
+#include <glib.h>
+#include "mongoose.h"
+
+typedef struct {
+	bool started;
+	bool finished;
+	bool success;
+	GString* file_path;
+	GString* srv_result_full_msg;
+} ws_verif_data;
+
+static const unsigned int websocket_single_poll_length_ms = 250;
+static const unsigned int websocket_max_polls_before_start = 60;
+
+static const char verifier_url[] = "ws://141.13.106.29:8080/engineverification/";
```

Comment:
Make this configurable from the `client.html`?

Reply:
Not trivial. Will be in second or third round of adjustments